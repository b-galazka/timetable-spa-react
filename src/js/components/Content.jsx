import React, { Component } from 'react';
import classNames from 'classnames';

import Sidebar from './Sidebar';
import Timetable from './Timetable';

import '../../scss/content.scss';
import '../../scss/animations.scss';

export default class Content extends Component {

    constructor() {

        super();

        this.state = {
            animationClass: true
        };

        this.removeAnimationClass = this.removeAnimationClass.bind(this);
    }

    render() {

        return (
            <div
                className={

                    classNames({
                        app__wrapper: true,
                        animated: this.state.animationClass
                    })
                }

                onAnimationEnd={this.removeAnimationClass}
            >
                <Sidebar />
                <Timetable />
            </div>
        );
    }

    removeAnimationClass() {

        this.setState({
            animationClass: false
        });
    }
}