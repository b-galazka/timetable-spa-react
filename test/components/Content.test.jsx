import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import Content from '../../src/js/components/Content';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../src/js/components/Sidebar', () => 'Sidebar');
jest.mock('../../src/js/components/Timetable', () => 'Timetable');

describe('Content component', () => {

    it('should render Sidebar and Timetable', () => {

        const tree = renderer.create(<Content />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should remove "animated" CSS class when animation is finished', () => {

        const content = mount(<Content />);

        content.find('.app__wrapper').simulate('animationEnd');

        expect(content.find('.app__wrapper').hasClass('animated')).toBe(false);
    });
});