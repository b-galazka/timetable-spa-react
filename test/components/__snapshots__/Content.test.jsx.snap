// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component should render Sidebar and Timetable 1`] = `
<div
  className="app__wrapper animated"
  onAnimationEnd={[Function]}
>
  <Sidebar />
  <Timetable />
</div>
`;
